src/app/code/community/Omise/Gateway/Model/OmiseCharge.php

```diff
@@ -1,6 +1,51 @@
 <?php
 class Omise_Gateway_Model_OmiseCharge extends Omise_Gateway_Model_Omise
 {
+
+    /**
+     * @param string $id
+     * @return OmiseTransfer|array
+     */
+    public function retrieveOmiseCharge($id)
+    {   
+    
+        try {
+            $charge = OmiseCharge::retrieve($id, $this->_public_key, $this->_secret_key);
+            return $charge;
+        } catch (Exception $e) {
+            return array('error' => $e->getMessage());
+        }
+    }
+
+    /**
+     * @param array $params
+     * @return OmiseTransfer|array
+     */
+    public function retrieveOmiseCharges($params)
+    {   
+        if(!isset($params['page']))
+            $params['page'] = 1;
+        
+        if(!isset($params['limit']))
+            $params['limit'] = 10;
+
+        try {
+
+            $check = OmiseCharge::retrieve('', $this->_public_key, $this->_secret_key);
```

Comment:
about `OmiseCharge::retrieve('', $this->_public_key, $this->_secret_key);`

I think if you add `?limit=1` it will be better for api side. Because it will reduce a query in the api side to get a records to you (extra point, performance in this page should be improve).